src/odemis/util/img.py

```diff
@@ -386,11 +386,42 @@ def DataArray2RGB(data, irange=None, tint=(255, 255, 255)):
     return rgb
 
 
+def ensureZYX(data, zIndex=0):
+    """
+    Extracts an XY plane from a ZYX image at the index given by zIndex (int)
+    Returns the data array, which is now 2D
+    """
+
+    d = data.view()
+    if len(d.shape) < 2:
+        d.shape = (1,) * (2 - len(d.shape)) + d.shape
+
+    elif len(d.shape) > 2:
+
+        pxs = d.metadata.get(model.MD_PIXEL_SIZE)
+        pos = d.metadata.get(model.MD_POS)
+        
+        if pxs is not None and len(pxs) == 3:
+            height = d.shape[0] * pxs[2]  # ZYX order
+            
+            d = d[zIndex]  # Remove z
```

Comment:
You should pick the zIndex + make sure there are no other higher dimensions unconditionally (ie, not inside the `if pxs...`). In the `if`, only put the code to handle the metadata.